### ISSUE TITLE
Help with getting Potlatch3 running

### ISSUE THREAD

**Issue by Russ-McD** [systemed/potlatch3#39]
After replacing my HDD and rebuilding the Windows 7 software on it, Im having troubles with Potlatch.
Installed the AIR from Harman, and then P3.  The prog opens but  won't display any OSM data, or backgrounds.  The initial Zoom is 2, but when finally zooming in to around Z12, a message appears to say it cant download images ... check the internet connection.  The default server is still showing.
All the Win updates are in now, along with Java and anything else I remembered I used.  Help !

**Comment by systemed** [comment]
I know absolutely nothing about Windows I'm afraid, but two guesses:

- Is there some sort of firewall setting you need to change? It sounds like all internet access by P3 is being blocked
- Is there a "Run this program in compatibility mode" setting that might work? Or opening the app in 32-bit mode or something?

**Comment by Russ-McD** [comment]
Thanks Richard,
You may well have something with the Firewall, and I will try to set an
exception and let you know.
Rgds.

**Comment by Russ-McD** [comment]
Nope, no joy... tried allowing through Firewall, tried various compatibility modes, tried run as admin .... nothing changes.

**Comment by BikePC** [comment]
Hi Russ, I run Potlatch.exe from the unpacked .zip archive "Potlatch_3_Windows_2022-01-24.zip" without separately installing Adobe AIR. Maybe that aproach may work for you too? Nothing done with firewall settings, or elsewhat - all Windows default. So uninstall AIR, unpack the .zip and run Potlatch.exe from the unpacked folder.

**Comment by Russ-McD** [comment]
Hi, thanks for the suggestion - I downloaded as suggested, unpacked into a
new P3 directory, ran the Potlatch exe file...
As usual, the Prog opens, but nothing is displayed (like backgrounds or map)

If I have to have a stab at whats going wrong, Im assuming the first time
the Prog is run, it should be asking me for an OAuth ... this never seems
to happen.

I have checked "My OAuth Details" and made sure there is no Potlatch in
there, so I assume it should be asking for one. I never get this.  In fact,
there is no login, so it has no idea who I am ... which is why I cant see
any of my GPS traces, I assume.

All a bit of a mystery to me.

Rgds, Russ

**Comment by BikePC** [comment]
Hi Russ, I vaguely remember having met something similar when installing Potlatch on a second Windows 10 PC here in the house. Possibly it's the order of doing things?
- First start your browser, open openstreetmap.org and **login**
- On openstreetmap.org zoom in to the area where you want to start editing (zoom 30 or 50 meters)
- Then run Potlatch.exe as you did above (from the unpacked .zip archive)
- Go back to openstreetmap.org in the browser window and click the rolldown arrow beside the EDIT button at the top of the screen and choose "Edit with Remote Control"
- The first time you select this, you will be prompted for your username (email address?) and password. This is from memory - it also can be that the login credentials are asked when you try to save your edit(s). Your credentials are asked only once. I've tried to reproduce the whole procedure on my laptop with a working Potlatch setup, but the login step in Potlatch does not come back. Where Potlatch (or the browser) stores the credentials I cannot find, so I'll have to start over on a clean machine to check a fresh install of Potlatch.

For your info: I'm running Firefox 115.0.3 as my default browser and use no stored login on openstreetmap.org (enter credentials on every login occasion.

Hope this may help you a bit further...
Rgds, Pim

**Comment by BikePC** [comment]
One more observation:
I changed the login password for openstreetmap.org and expected that Potlatch would ask for my new password at opening or saving a new changeset, but none of that. Everything kept working as normal. In my opinion this means that Potlatch, once handshaked with a logged-in openstreetmap.org uses the credentials from OSM.

**Comment by Russ-McD** [comment]
I'll give it a try later, but have to say that being unable to use P3 for
so long, has forced me to learn the  iD editor, which does seem to be
better in many respects.  At least one the learning curve has been climbed !
Rgds.
